SerialPrograms/Source/Integrations/DiscordSocial/DiscordSocial.cpp

```diff
@@ -0,0 +1,171 @@
+#ifdef PA_SOCIAL_SDK
+#define DISCORDPP_IMPLEMENTATION
+
+#include <thread>
+#include "DiscordSocial.h"
+#include "CommonFramework/Logging/Logger.h"
+#include "Integrations/ProgramTracker.h"
+#include "../../CommonFramework/GlobalSettingsPanel.h"
+#include "../../CommonFramework/Globals.h"
+
+using namespace discordpp;
+namespace PokemonAutomation{
+namespace Integration{
+namespace DiscordSocialSDK{
+
+DiscordSocial& DiscordSocial::instance(){
+    static DiscordSocial instance;
+    return instance;
+}
+
+void DiscordSocial::run(){
+    m_client = std::make_shared<Client>();
+    if (!m_client){
+        log("Failed to initialize DiscordSocialSDK.", "run()", LoggingSeverity::Error);
+        m_client.reset();
+        return;
+    }
+
+    m_client->SetApplicationId(m_app_id);
+    m_client->AddLogCallback([&](auto message, auto severity){
+        log(message, "Internal", severity);
+    }, m_log_level);
+
+    std::thread(&DiscordSocial::thread_loop, this).detach();
+}
+
+void DiscordSocial::thread_loop(){
+    log("Starting Discord Rich Presence update thread...", "thread_loop()", LoggingSeverity::Info);
+    m_running.store(true, std::memory_order_release);
+    m_activity = create_activity();
+    m_client->UpdateRichPresence(m_activity, [&](ClientResult result){
+        if (!result.Successful()){
+            log("Rich Presence initial update failed.", "thread_loop()", LoggingSeverity::Warning);
+        }
+    });
+
+    while (m_running.load(std::memory_order_relaxed)){
+        try{
+            if (!m_running.load(std::memory_order_relaxed) || !GlobalSettings::instance().RICH_PRESENCE){
+                break;
+            }
+
+            RunCallbacks();
+            update_rich_presence();
+            std::this_thread::sleep_for(std::chrono::seconds(1));

```

Comment:
Do we need to do this every second? Is it possible to diff to see if anything changed. And only update if something changed?

Ideally CC itself should push these changes out, but now is not the time to re-architect that part so this is fine.